primer/react
# test(react): remove support for testing React 17

Starting in v36, we will support only React 18. This PR removes these checks in current `main` as we look to merge in `next-major` over the next couple of weeks 👀 

### Changelog



#### New



#### Changed



#### Removed



- Remove test and type-check matrices in CI workflow
- Remove `REACT_VERSION_LATEST` and `REACT_VERSION_17` flags from Jest config and tests

### Rollout strategy



We will need to update the protected branch for `main` to expect `test` and `type-check` instead of the 17 and 18 suffixed jobs that we currently have. Once that's complete, we can merge in the PR and update all other PRs to inherit the new workflow jobs.

### Testing & Reviewing



- Verify that the test and type-check jobs appear and pass
- Verify that the test and type-check jobs with (17) and (18) appended to them do not run
